### PR TITLE
Support Include in AST

### PR DESCRIPTION
**Pull request by lucifer1004**
This PR contains the AST part of #762

### CODE REVIEW

**Comment by lucifer1004** [comment]
@certik This PR contains the AST part of #762.

To test this, I added a `--no-prescan` CLI option.

File: run_tests.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def is_included(backend):
             filename,
             update_reference,
             extra_args)
-    if ast:
+    if ast:
```
Comment (certik) [inline run_tests.py:63]:
```suggestion
    if ast:
```

**Comment by certik** [comment]
Thanks! I think this looks good, I left some minor change above. I will test it carefully tomorrow.

I think we'll eventually make `--no-prescan` the default, and add `--prescan` as an option.

**Comment by lucifer1004** [comment]
We need to figure out how to fix the nbconvert issue in CI. It is really annoying.

**Comment by certik** [comment]
It's the issue #584, yes, we have to fix it.

**Comment by lucifer1004** [comment]
Can we merge this? @certik

**Comment by certik** [comment]
I need to play with this by hand tomorrow, I'll merge it if everything works.

**Comment by lucifer1004** [comment]
> I need to play with this by hand tomorrow, I'll merge it if everything works.

That's fine.

**Comment by certik** [comment]
I made the tests working with prescanner, so LFortran and GFortran can actually compile them.

I added an error at the semantic level with instructions how to get include working:
```console
$ lfortran --no-prescan no_prescan_include2.f90
L: 21
L: 43
semantic error: Include statement is not implemented at the AST level yet. You have to run LFortran with prescanning which can handle include statements.
 --> no_prescan_include2.f90:1:1
  |
1 | program include2
  | ^ Enable prescanner to handle this


Note: if any of the above error or warning messages are not clear or are lacking
context please report it to us (we consider that a bug that needs to be fixed).
```
I noticed the location information is wrong. So I tested it with:
```diff
--- a/src/lfortran/semantics/ast_common_visitor.h
+++ b/src/lfortran/semantics/ast_common_visitor.h
@@ -854,6 +854,8 @@ public:
     }
 
     void visit_Include(const AST::Include_t &x) {
+        std::cout << "L: " << x.base.base.loc.first << std::endl;
+        std::cout << "L: " << x.base.base.loc.last << std::endl;
         diag.semantic_error_label(
             "Include statement is not implemented at the AST level yet. You have to run LFortran with prescanning which can handle include statements.",
             {x.base.base.loc},
```
And it seems the location info in AST is correct. So it's some bug, probably due to not using prescanner so the remapping is probably not initialized properly. That's a separate bug.

So I think this is good to merge now.

Thanks!

**Comment by certik** [comment]
I also reworked the tests to use the usual infrastructure.

**Comment by certik** [comment]
Not sure why the Windows test failed, rerunning.

**Pull request merged**